src/main/java/pl/tsimur/jpatest/service/UserServiceImpl.java

```diff
@@ -3,23 +3,36 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import pl.tsimur.jpatest.model.Subscription;
 import pl.tsimur.jpatest.model.Tariff;
 import pl.tsimur.jpatest.model.User;
+import pl.tsimur.jpatest.repository.SubscriptionRepository;
 import pl.tsimur.jpatest.repository.UserRepository;
 
 import java.util.List;
+import java.util.Optional;
 
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
 
     private final UserRepository userRepository;
+    private final SubscriptionRepository subscriptionRepository;
 
     @Override
-    public void buySubscription(Integer id, Tariff tariff) {
-//    TODO
+    public void buySubscription(Integer id, Tariff tariff) throws Throwable {
+        Optional foundUser = userRepository.findById(id);
```

Comment:
```suggestion
        User buyer = userRepository.findById(id)
                .orElseThrow(()-> new NullPointerException( "User with id = "+id+ "didn't find" ));
```
Этот тип всегда с ромбовыми скобками, где указываем тип`Optional<User>` - тебе идея должна была светить и нажать надо было:

- `OPTION` + `ENTER`
 
тогда бы подставило само.

Но я не стал заводить отдельную переменную под это в моем suggestion

Reply:
Поменял на Enum и удалил Throwable с сигнатуры